lib/wasm/WebAssemblyGenerator.js

```diff
@@ -181,13 +189,63 @@ const rewriteImportedGlobals = state => bin => {
 	return addWithAST(state.ast, bin, newGlobals);
 };
 
+/**
+ * Rewrite the export names
+ * @param {Object} state state
+ * @param {Object} state.ast Module's ast
+ * @param {Object} state.module Module
+ * @returns {ArrayBufferTransform} transform
+ */
+const rewriteExportNames = ({ ast, module }) => bin => {
+	return editWithAST(ast, bin, {
+		ModuleExport(path) {
+			const usedName = module.isUsed(path.node.name);
+			if (usedName) {
+				path.node.name = usedName;
+				// TODO remove this when fixed in @webassemblyjs
+				path.node.descr.id = t.numberLiteral(+path.node.descr.id.raw);
+			} else {
+				path.remove();
+			}
+		}
+	});
+};
+
+/**
+ * Mangle import names and modules
+ * @param {Object} state state
+ * @param {Object} state.ast Module's ast
+ * @param {Map<string, UsedWasmDependency>} state.usedDependencyMap mappings to mangle names
+ * @returns {ArrayBufferTransform} transform
+ */
+const rewriteImports = ({ ast, usedDependencyMap }) => bin => {
+	return editWithAST(ast, bin, {
+		ModuleImport(path) {
+			const result = usedDependencyMap.get(
+				path.node.module + ":" + path.node.name
+			);
+			if (result === undefined) {
+				path.remove();
+			} else {
+				path.node.module = WebAssemblyUtils.MANGLED_MODULE;
+				path.node.name = result.name;
+				if (path.node.descr.id)
+					path.node.descr.id = t.numberLiteral(+path.node.descr.id.raw);
+				if (path.node.descr.name)
+					path.node.descr.name = t.numberLiteral(+path.node.descr.name.raw);
```

Comment:
i'm not sure we need to rename the `path.node.descr`,  it's only the internal reference and uses an index anyway.